File: Composicion_y_Agregacion/Ejercicio3/Java/Avion.java
package Composicion_y_Agregacion.Ejercicio3.Java;

public class Avion { 
    private String modelo;
    String fabricante; 
    Parte[] partes;
    int cont;

    public Avion (String modelo, String fabricante, int capacidad) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new Parte[capacidad]; 
        cont = 0;
    }

    void agregar_parte (Parte p) {
        if (cont< partes.length) {
            partes[cont] = p;
            cont++; }
        else { System.out.println("No se pueden agregar más partes al avión."); 
    }
}

    void mostrar_avion(){
        System.out.println("\nModelo: " + modelo); 
        System.out.println("Fabricante: " + fabricante); 
        System.out.println("\n----Partes del avión:----"); 
        for (int i = 0; i < cont; i++) {
            partes[i].mostrar_info();
    }
        
    }   

    String getModelo() { return modelo; } 
    String getFabricante() { return fabricante; }
}

File: Composicion_y_Agregacion/Ejercicio3/Python/Avion.py

from Parte import Parte
class Avion:
    def __init__(self, modelo, fabricante, capacidad):
        self.modelo = modelo
        self.fabricante = fabricante
        self.partes = [ ]        
        self.capacidad = capacidad
    def agregar_parte(self, p):
        if len(self.partes) < self.capacidad:
            self.partes.append(p)
        else:
            print("No se pueden agregar más partes al avión.")
            
    def mostrar_avion(self):
        print(f"\nModelo: {self.modelo}")
        print(f"Fabricante: {self.fabricante}")
        print("\n----Partes del avión:----")
        for parte in self.partes:
            parte.mostrar_info()
            
    def get_modelo(self):  return self.modelo
    def get_fabricante(self):  return self.fabricante

